internal/db/dao.go

```diff
@@ -97,6 +98,25 @@ func (dao *PostgresDAO) GetGoals(pagination Pagination, filter GetGoalsFilter) (
 	return list, nil
 }
 
+func (dao *PostgresDAO) GetGoal(id string) (top90.Goal, error) {
+
+	query := fmt.Sprintf("SELECT * FROM %s WHERE id = %s", tableNames.Goals, id)
+
+	var video top90.Goal
+	row, err := dao.DB.Query(query)
```

Comment:
This is still not working correctly. We have to use different syntax since we are querying for one row instead of multiple rows (like in GetGoals). So this whole function should be
```
query := fmt.Sprintf("SELECT * FROM %s WHERE %s = $1", tableNames.Goals, goalColumns.Id)

var goal top90.Goal
row := dao.DB.QueryRow(query, id)

err := row.Scan(&goal.Id, &goal.RedditFullname, &goal.RedditLinkUrl, &goal.RedditPostTitle, &goal.RedditPostCreatedAt, &goal.S3ObjectKey, &goal.CreatedAt)
if err != nil {
    return goal, err
}

return goal, nil
```

With that, it will work.

Reply:
ok, was trying that. Thank you.